app/src/main/java/com/infomaniak/drive/views/FileInfoActionsView.kt

```diff
@@ -454,52 +447,75 @@ class FileInfoActionsView @JvmOverloads constructor(
 
     interface OnItemClickListener {
 
-        val ownerFragment: Fragment
-        val currentFile: File
+        val ownerFragment: Fragment?
+        val activity: Activity?
+        val currentFile: File?
 
-        private val context: Context get() = ownerFragment.requireContext()
+        private val context: Context get() = ownerFragment?.requireContext() ?: activity!!
 
         private fun trackFileActionEvent(name: String, value: Boolean? = null) {
             context.trackFileActionEvent(name, value = value?.toFloat())
         }
 
+        fun shareFile() = trackFileActionEvent("shareFile")
```

Comment:
There's already a event being tracked when "sending a copy" of a file inside your `Context.shareFile()`. Why put a second new and different one here? The old one looks good enough to me. 

Also because you're not using the interface in the second place where `Context.shareFile()` is called, this "shareFile" event is only triggered for external PDFs. Just so you know, instead of further trying to improve the interface which is obviously not very well thought out as of right now, you could simply call your `Context.shareFile()` in your new Fragment the same way it was already called in the old Fragment I think

Reply:
I replaced the tracking in the default implementation and just use the one in the extension.